drivers/gpio/gpio_stm32.c

```diff
@@ -211,47 +311,99 @@ const int gpio_stm32_enable_int(int port, int pin)
 	clock_control_on(clk, (clock_control_subsys_t *) &pclken);
 #endif
 
-	uint32_t line;
+	if (pin > 15) {
+		return -EINVAL;
+	}
 
+	gpio_stm32_set_exti_source(port, pin);
+
+	return 0;
+}
+
+/**
+ * @brief Get enabled GPIO port for EXTI of the specific pin number
+ */
+static int gpio_stm32_int_enabled_port(int pin)
```

Comment:
The name of the function doesn't make it clear what the function does. It may be better to use `gpio_stm32_get_exti_source` directly (after moving there the check for an invalid pin).

Reply:
> (after moving there the check for an invalid pin).

Would be a relief to have this done in API.